tests/test_server.py

```diff
@@ -133,19 +133,24 @@ def test_bookmark_api(client):
     assert rd.status_code == 200
     assert rd.get_json() == {
         'description': '', 'tags': [], 'title': 'Google', 'url': 'http://google.com'}
-    rd = client.put('/api/bookmarks/1', data={'tags': [',tag1,tag2,']})
+    rd = client.put('/api/bookmarks/1', json={'tags': ['tag1', 'tag2']})
+    assert rd.status_code == 400
```

Comment:
This request shouldn't fail tho.

…You do realize that POST and PUT requests should have different validation rules, right?

Reply:
REST isn't a strict guideline; there are variations between different implementations.

This particular API has no use for full-replacement-updates (and frankly speaking they're rarely needed in the first place), so the PUT method is used for partial updates instead (which is fairly common BTW); you could tell that by the previous test, if anything.